tests/test_mixing.py

```diff
@@ -180,8 +181,24 @@ def test_mix_proportion_range(root: str):
         streams.append(stream)
     dataset = StreamingDataset(streams=streams, samples_per_epoch=12)
     assert dataset.num_samples == 8
-    assert walk(dataset) == [2, 3, 4, 4, 5, 5, 6, 6, 6, 7, 7, 7]
+    assert walk(dataset) == [2, 3, 4, 5, 4, 5, 6, 7, 6, 7, 6, 7]
     for i, stream in enumerate(streams):
         assert float_eq(stream.proportion, i / 6)
         assert stream.repeat == i
         assert stream.samples == i * 2
+
+
+def test_mix_balance(root: str):
```

Comment:
You can use `local_remote_dir` pytest annotation which deletes the directory when test finishes. checkout `test_writer.py` for example

Reply:
done